Models/VanillaConsumable.cs

```diff
@@ -10,6 +10,11 @@ public VanillaConsumable(ConsumableItem item) {
 		}
 
 		public ConsumableType Type => Item switch {
```

Comment:
This whole class has alot of hardcode which I don't need is necessary. You can compress the DisplayText to just `Lang.GetItemName(vanillaItemId)`.
Not sure about the HoverText since you add some extra stuff to it. In general, any strings you are using that should be displayed to users are supposed to go into the localization file (like the config does at the moment). You can read more about how to add localization to your mod here: <https://github.com/tModLoader/tModLoader/wiki/Localization>.
ConsumableItem -> ItemId is backwards design imo, it should be sufficient to just have them referenced as the ItemId directly as it's vanilla, and use the consts (`ItemID.LifeCrystal` instead of 29 etc.)
My impression is that all of these could be replaced by using the `Consumable` class which you already have. You can use the API in your own mod aswell!